test/test_config.js

```
@@ -159,6 +163,177 @@ describe("SplunkLogger", function() {
+
+            assert.ok(logger);
+            assert.ok(logger._timerID);
+
+            assert.strictEqual(config.token, logger.config.token);
+            assert.strictEqual("splunk-javascript-logging/0.8.0", logger.config.name);
+            assert.strictEqual("localhost", logger.config.host);
+            assert.strictEqual("/services/collector/event/1.0", logger.config.path);
+            assert.strictEqual("https", logger.config.protocol);
+            assert.strictEqual("info", logger.config.level);
+            assert.strictEqual(true, logger.config.autoFlush);
+            assert.strictEqual(100, logger.config.batchInterval);
+            assert.strictEqual(8088, logger.config.port);
+            assert.strictEqual(0, logger.config.maxRetries);
+        });
+        it("should not set a batch interval timer with autoFlush on, & default batchInterval", function() {
+            var config = {
+                token: "a-token-goes-here-usually",
+                autoFlush: true
+            };
+            var logger = new SplunkLogger(config);
+
+            assert.ok(logger);
+            assert.ok(!logger._timerID);
+
+            assert.strictEqual(config.token, logger.config.token);
+            assert.strictEqual("splunk-javascript-logging/0.8.0", logger.config.name);
+            assert.strictEqual("localhost", logger.config.host);
+            assert.strictEqual("/services/collector/event/1.0", logger.config.path);
+            assert.strictEqual("https", logger.config.protocol);
+            assert.strictEqual("info", logger.config.level);
+            assert.strictEqual(true, logger.config.autoFlush);
+            assert.strictEqual(0, logger.config.batchInterval);
+            assert.strictEqual(8088, logger.config.port);
+            assert.strictEqual(0, logger.config.maxRetries);
+        });
+        it("should not set a batch interval timer with autoFlush off, & batchInterval set", function() {
```

I wonder if this case (auto flush off with any kind of batching) should throw an error.